components/eamxx/docs/developer/dev_testing/test_all_eamxx.md

```diff
@@ -0,0 +1,304 @@
+                            Just (incremental) build and test. (default: False)
+      --quick-rerun-failed  Do not clean the build dir, and do not reconfigure.
+                            Just (incremental) build and retest failed tests only.
+                            (default: False)
+      --make-parallel-level MAKE_PARALLEL_LEVEL
+                            Max number of jobs to be created during compilation.
+                            If not provided, use default for given machine.
+                            (default: 0)
+      --ctest-parallel-level CTEST_PARALLEL_LEVEL
+                            Force to use this value for CTEST_PARALLEL_LEVEL. If
+                            not provided, use default for given machine. (default:
+                            0)
+      -x, --extra-verbose   Have ctest run in extra-verbose mode, which should
+                            print full output from all tests (default: False)
+      --limit-test-regex LIMIT_TEST_REGEX
+                            Limit ctest to running only tests that match this
+                            regex (default: None)
+      --test-level {at,nightly,experimental}
+                            Set the testing level. (default: at)
+      --test-size {short,medium,long}
+                            Set the testing level. Defaults to medium unless the
+                            test is cov or mem_check(short is default in those
+                            cases). (default: None)
+
+    ```
+<!-- markdownlint-enable MD013 -->
+
+If you are unsure of the CMake configuration for you development cycle, one
+trick you can use is to run `test-all-eamxx --config-only`
+for the `dbg` test and then copy the `cmake` commands from the resulting output.
+
+```shell
+cd $eamxx_repo/components/eamxx
+./scripts/test-all-eamxx -t dbg -m $machine
+# wait for a few seconds*
+# Ctrl-C *
+# Copy the contents of DCMAKE_COMMAND that was passed to ctest *
+# Add "cmake" to beginning of contents and path to eamxx at the end. *
+```
+
+Considerations for using `test-all-eamxx`:
+
+- Your machine must be known to our scripts, see above.
+- If you try to run commands by-hand (i.e., outside of `test-all-eamxx`,
+`cmake`, `make`, `ctest`, etc.), you'll need to
+load the eamxx-env into your shell, which can be achieved by running
+`cd eamxx/scripts; eval $(./eamxx-env-cmd $machine)`
+- `test-all-eamxx` expects to be run from a compute node if you
+are on a batch machine.
+- You'll need to think about your baseline situation, as many of our
+  tests rely on pre-existing baselines. The `-b` option controls the baseline
+  location and can have the following values:
+      - `AUTO`
+          - A common public baseline area shared by all developers
+      - `LOCAL`
+          - A private baseline area for the current developer in the current repo
+      - `$path`
+          - A specific arbitrary path
+      - `None`
```

Comment:
I think this may be misread. Listing `None` may be read as `-b None`, but doing that is interpreted as the string "None", rather than the python keyword `None`... I would just not add this item, and move the line below (about lack of -b) above the itemized list.